src/diffpy/labpdfproc/tests/test_tools.py

```diff
@@ -241,3 +243,65 @@ def test_load_user_metadata_bad(inputs, msg):
     actual_args = get_args(cli_inputs)
     with pytest.raises(ValueError, match=msg[0]):
         actual_args = load_user_metadata(actual_args)
+
+
+# For each test case, check username and email are properly loaded and written into config file
+params_user_info = [
+    # No existing config file, user enters valid username and email
+    (["good_username", "good@email.com", "non_existing_file.json"], ["good_username", "good@email.com"]),
+    # There exists a config file, so we read it and fetch info, no user input is fine
+    (["", "", "test.json"], ["good_username", "good@email.com"]),
+    # There exists a config file, user input username/email overwrites the username/email in the file
+    (["new_username", "", "test.json"], ["new_username", "good@email.com"]),
+    (["", "new@email.com", "test.json"], ["good_username", "new@email.com"]),
+    (["new_username", "new@email.com", "test.json"], ["new_username", "new@email.com"]),
+    # There exists a config file that does not contain username/email, then we write user inputs into config file
+    (["good_username", "good@email.com", "test2.json"], ["good_username", "good@email.com"]),
+]
+
+
+@pytest.mark.parametrize("inputs, expected", params_user_info)
+def test_load_user_info(monkeypatch, inputs, expected, user_filesystem):
+    os.chdir(user_filesystem / "conf_dir")
+    expected_username, expected_email = expected
+    input_username, input_email, input_config_file = inputs
```

Comment:
we are not inputting a config file.  options are:

2. find `diffpyconfig.json` in `cwd`
3. failing that, find `diffpyconfig.json` in `~`
4. failing that, check if username and email are provided in the input line
5. failing that, trigger the workflow to create `diffpyconfig.json` in `~`

Then the workflow is:

1. If 1 or 3, check that those files contain username/email.  If they do continue, if they don't trigger workflow to update `diffpyconfig.json`
2. If continue above, then check if username/email are provided at the command line.  If yes, use them (but don't update config).  If not,  use the values in config.
3. if (4.) above, then use these values to create the `diffpyconfig.json`

Reply:
I edited the tests. I included "paths" in @pytest.mark.parametrize because it helps manage different cwd and home directories, not sure if this is the best approach though.